src/wix/WixToolset.Core.Burn/Bundles/ProcessBundlePackageCommand.cs

```diff
@@ -0,0 +1,152 @@
+
+    /// <summary>
+    /// Initializes package state from the Bundle contents.
+    /// </summary>
+    internal class ProcessBundlePackageCommand
+    {
+        public ProcessBundlePackageCommand(IServiceProvider serviceProvider, IntermediateSection section, PackageFacade facade, Dictionary<string, WixBundlePayloadSymbol> packagePayloads, string intermediateFolder)
+        {
+            this.Messaging = serviceProvider.GetService<IMessaging>();
+            this.BackendHelper = serviceProvider.GetService<IBackendHelper>();
+            this.PackagePayloads = packagePayloads;
+            this.Section = section;
+            this.Facade = facade;
+            this.IntermediateFolder = intermediateFolder;
+        }
+
+        private IMessaging Messaging { get; }
+
+        private IBackendHelper BackendHelper { get; }
+
+        private Dictionary<string, WixBundlePayloadSymbol> PackagePayloads { get; }
+
+        private PackageFacade Facade { get; }
+
+        private IntermediateSection Section { get; }
+
+        private string IntermediateFolder { get; }
+
+        public List<ITrackedFile> TrackedFiles { get; } = new List<ITrackedFile>();
+
+        /// <summary>
+        /// Processes the Bundle packages to add properties and payloads from the Bundle packages.
+        /// </summary>
+        public void Execute()
+        {
+            var bundlePackage = (WixBundleBundlePackageSymbol)this.Facade.SpecificPackageSymbol;
+            var packagePayload = this.PackagePayloads[this.Facade.PackageSymbol.PayloadRef];
+            var sourcePath = packagePayload.SourceFile.Path;
+
+            using (var burnReader = BurnReader.Open(this.Messaging, sourcePath))
+            {
+                if (burnReader.Invalid)
+                {
+                    return;
+                }
+
+                var baFolderPath = Path.Combine(this.IntermediateFolder, burnReader.BundleId.ToString());
+
+                if (!burnReader.ExtractUXContainer(baFolderPath, baFolderPath))
+                {
+                    return;
+                }
+
+                foreach (var filePath in Directory.EnumerateFiles(baFolderPath, "*.*", SearchOption.AllDirectories))
+                {
+                    this.TrackedFiles.Add(this.BackendHelper.TrackFile(filePath, TrackedFileType.Temporary, packagePayload.SourceLineNumbers));
+                }
+
+                switch (burnReader.MachineType)
+                {
+                    case BurnCommon.IMAGE_FILE_MACHINE_ARM:
+                    case BurnCommon.IMAGE_FILE_MACHINE_ARMNT:
+                    case BurnCommon.IMAGE_FILE_MACHINE_I386:
+                        break;
+                    case BurnCommon.IMAGE_FILE_MACHINE_AMD64:
+                    case BurnCommon.IMAGE_FILE_MACHINE_ARM64:
+                        bundlePackage.Win64 = true;
+                        break;
+                    default:
+                        Debug.Assert(false, "Unknown machine type");
```

Comment:
Since `Debug.Assert()` disappears in Release builds, I generally recommend throwing `InvalidOperationException` or `ArgumentException` or something with the error message so that we likely get a bug if/when it happens.

Reply:
I'm really not a fan of blocking people from building a bundle because one of the packages couldn't be harvested. It's looking like Burn's going to ignore this anyway until/unless it actually starts verifying that the bundle it found in the registry was the right bitness.

Reply:
Maybe a warning then so the dev has some indication that harvesting didn't behave normally. I'm concerned about the future if this list changes and it silently slips through rather than noisly failing so we can find and fix it.

Reply:
I guess. I don't imagine Windows or WiX supporting additional machine types anytime soon.

Reply:
I would have said the same thing before ARM came along. :)  But I've been bitten by silent `Debug.Assert()`s enough that I don't care for them (unless they are there to provide useful developer context for a crash that is coming when violated).  :)

Reply:
I mean, ARM/ARM64 has been a widely used architecture for over 10 years and there's still no stable WiX bundle for it. To be fair Windows hasn't supported for that long, but still.